bokeh/models/widgets/inputs.py

```diff
@@ -89,48 +89,49 @@ def coerce_value(cls, val):
 #-----------------------------------------------------------------------------
 
 class FileInput(Widget):
-    ''' Present a file-chooser dialog to users and return the contents of a
-    selected file.
-
+    ''' Present a file-chooser dialog to users and return the contents of the 
+    selected files 
     '''
-
-    value = String(default="", readonly=True, help="""
-    A base64-encoded string of the contents of the selected file.
-    """)
-
-    mime_type = String(default="", readonly=True, help="""
-    The mime type of the selected file.
-    """)
-
-    filename = String(default="", readonly=True, help="""
-    The filename of the selected file.
-
-    .. note::
-        The full file path is not included since browsers will not provide
-        access to that information for security reasons.
-    """)
+    
+    value = Either(String, List(String), default='', readonly=True, help='''
+    if multiple=False or ommitted, base64-encoded string of the contents of the 
+    selected file,
+    if multiple=True or 'Multiple' a list of base64 encoded strings of the contents 
+    of selected files in the order given by filename
+    ''')
+
+    mime_type = Either(String, List(String), default='', readonly=True, help='''
+    if multiple=False or ommitted, string with the mime-type of the selected file,
+    if multiple=True or 'Multiple' a list of strings with the mime-types of the 
+    selected files in the order given by filename
+    ''')
+
+    filename = Either(String, List(String), default='', readonly=True, help='''
+    if multiple=False or ommitted, the filename of the selected file as string,
+    if multiple=True or 'Multiple' a list of strings giving the filenames of the 
+    selected files 
+    ''')
 
     accept = String(default="", help="""
     Comma-separated list of standard HTML file input filters that restrict what
     files the user can pick from. Values can be:
-
     `<file extension>`:
         Specific file extension(s) (e.g: .gif, .jpg, .png, .doc) are pickable
-
     `audio/*`:
         all sound files are pickable
-
     `video/*`:
         all video files are pickable
-
     `image/*`:
         all image files are pickable
-
     `<media type>`:
         A valid `IANA Media Type`_, with no parameters.
-
```

Comment:
@bruns01 Sphinx is very picky and fussy. The whitespace lines here is actually meaningful and necessary for Sphinx to generate the rendered HTML docs correctly.

Reply:
Reapplied the empty lines.
(I don't remember to have them intentionally removed in the first place (???).)